packages/angular/projects/angular/src/internal/dynamic-template/dynamic-template.component.ts

```
@@ -0,0 +1,99 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ComponentRef,
+  EmbeddedViewRef,
+  Input,
+  TemplateRef,
+  ViewContainerRef,
+} from '@angular/core';
+import { UnknownRecord } from '@devextreme/core';
+import { AngularTemplate } from '../types';
+import { ViewComponent } from './view.component';
+
+interface AngularTemplateContext<T> {
+  $implicit: T;
+}
+
+@Component({
+  selector: 'dx-dynamic-template',
+  template: '',
+  styles: [':host {display: none; }'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DynamicTemplateComponent<
+  TComponent extends ViewComponent,
+  TData extends UnknownRecord,
+  > {
+  @Input() set template(template: AngularTemplate<TComponent>) {
+    if (template === this.templateValue) {
+      return;
+    }
+
+    this.setTemplate(template);
+  }
+
+  @Input() set data(data: TData) {
+    if (data === this.dataValue) {
+      return;
+    }
+
+    this.setTemplateData(data);
+  }
+
+  private templateValue?: AngularTemplate<TComponent>;
+
+  private dataValue?: TData;
+
+  private componentRef?: ComponentRef<TComponent>;
+
+  private templateRef?: EmbeddedViewRef<AngularTemplateContext<TData>>;
+
+  constructor(
+    private viewContainerRef: ViewContainerRef,
+  ) {
+  }
+
+  private setTemplate(template: AngularTemplate<TComponent>): void {
+    if (!template) {
```

Question: Is it OK that we doesn't reset templateValue it template is null or undefined?

---

We don't need the dynamic template component without a template.

If we want to destroy a template we can just destroy the dynamic-template component :)

E.g
```html
<dx-dynamic-template
        *ngIf="labelTemplate" <!-- <-- component will be destroyed if the 'labelTemplate' will be 'null or 'undefined' -->
        [template]="labelTemplate"
        [data]="{ label: label }">
      </dx-dynamic-template>
```

I will add this check to the radio-button component. Thanks!

---

Minor: Another worlds/ After your update template has never will null and if (!template) should raise assertion or debug warning about missing *ngIf="labelTemplate"?